src/v/cluster/data_migration_frontend.cc

```diff
@@ -120,6 +127,68 @@ bool frontend::data_migrations_active() const {
            && _cloud_storage_api_initialized;
 }
 
+ss::future<frontend::list_unmounted_topics_result>
+frontend::list_unmounted_topics() {
+    if (!data_migrations_active()) {
+        return ssx::now<list_unmounted_topics_result>(errc::feature_disabled);
+    }
+
+    vlog(dm_log.debug, "listing unmounted topics");
+
+    return container().invoke_on(
```

Comment:
do we want this to live in frontend and have a direct dependency to cloud storage api? @bashtanov

Reply:
Definitely not in backend, as there's no reconciliation/replication needed across nodes. I don't think it's worth creating a separate class for it, so I don't mind it being here.

Reply:
I'm wondering why we need this to be in data migrations code at all, vs e.g. some new `mount_manifest_downloader` class or some method in `cloud_storage` that has a `remote` as input. Then you can call that directly from the admin endpoint, or from this frontend if we need it.

Reply:
I moved this to topic_mount_handler. That's the right place for it. I still let admin access it through data_migrations_frontend for the purpose of encapsulating/hiding some lower level details/api surface.